src/rust/ide/enso-protocol/src/file_manager.rs

```diff
@@ -0,0 +1,411 @@
+//! Client library for the JSON-RPC-based File Manager service.
+
+#![warn(missing_docs)]
+#![warn(trivial_casts)]
+#![warn(trivial_numeric_casts)]
+#![warn(unused_import_braces)]
+#![warn(unused_qualifications)]
+#![warn(unsafe_code)]
+#![warn(missing_copy_implementations)]
+#![warn(missing_debug_implementations)]
+
+use crate::prelude::*;
+
+use crate::common::UTCDateTime;
+
+use json_rpc::api::Result;
+use json_rpc::Handler;
+use json_rpc::make_arg;
+use json_rpc::make_param_map;
+use json_rpc::make_rpc_methods;
+use futures::Stream;
+use serde::Serialize;
+use serde::Deserialize;
+use std::future::Future;
+use uuid::Uuid;
+
+
+
+// =============
+// === Event ===
+// =============
+
+/// Event emitted by the File Manager `Client`.
+pub type Event = json_rpc::handler::Event<Notification>;
+
+
+
+// ============
+// === Path ===
+// ============
+
+/// Path to a file.
+#[derive(Clone,Debug,Display,Eq,Hash,PartialEq,PartialOrd,Ord)]
+#[derive(Serialize, Deserialize)]
+#[derive(Shrinkwrap)]
+pub struct Path(pub String);
+
+impl Path {
+    /// Wraps a `String`-like entity into a new `Path`.
+    pub fn new(s:impl Str) -> Path {
+        Path(s.into())
+    }
+}
+
+
+
+// ====================
+// === Notification ===
+// ====================
+
+/// Notification generated by the File Manager.
+#[derive(Clone,Debug,PartialEq)]
+#[derive(Serialize, Deserialize)]
+#[serde(tag="method", content="params")]
+pub enum Notification {
+    /// Filesystem event occurred for a watched path.
+    #[serde(rename = "filesystemEvent")]
+    FilesystemEvent(FilesystemEvent),
+}
+
+
+
+// =======================
+// === FilesystemEvent ===
+// =======================
+
+/// Filesystem event notification, generated by an active file watch.
+#[derive(Clone,Debug,PartialEq)]
+#[derive(Serialize, Deserialize)]
+pub struct FilesystemEvent {
+    /// Path of the file that the event is about.
+    pub path : Path,
+    /// What kind of event is it.
+    pub kind : FilesystemEventKind
+}
```

Comment:
This doc doesnt tell anything. I know that field `FilesystemEvent::path` is "file system event path" and `FilesystemEvent::kind` is "file system event kind". Either make docs which add information or use allow missing docs.

Reply:
I remove kind docs. The documentation for `path` property makes sense for me.